spec/tasks/supprime_controle_campagne_spec.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+describe 'nettoyage:supprime_controle_campagne' do
+  include_context 'rake'
+
+  let(:logger) { RakeLogger.logger }
+
+  before { allow(logger).to receive :info }
+
+  context "d'une campagne vide" do
+    let(:campagne) { create :campagne }
+
+    it do
+      subject.invoke
+      expect(campagne.situations.count).to eql(0)
+    end
+  end
+
+  context 'une campagnes avec controle' do
+    let(:campagne) { create :campagne }
+    let(:situation) { create :situation_controle }
+    let!(:situations_configurations) do
+      create :situation_configuration, campagne_id: campagne.id, situation: situation
+    end
+
+    it do
+      expect(campagne.situations.count).to eql(1)
+      subject.invoke
+      expect(Campagne.first.situations.count).to eql(0)
```

Comment:
je trouve ça dommage de ne pas réutiliser l'object `campagne` du setup.
`expect(campagne.reload.situations.count).to eql(0)`

Reply:
Je ne connaissait pas `reload`. Allons y